### PR TITLE
feat: add unlock protocol

### LINKED ISSUE #2363: [Bounty] Unlock the imagination of online publication directly on Twitter with Unlock Protocol
## Catalogue

- Platform
  - [x] Gitcoin
  - [ ] Hackerlink
  - [ ] Mask
- Experience
  - [ ] Beginner
  - [x] Intermediate
  - [ ] Advanced
- Category
  - [ ] Mask Core Feature
  - [x] Mask Plugin
  - [ ] Security/Cryptography Feature
  - [ ] Documentation
  - [ ] Design
  - [ ] Other
    - [ ] /_List it here_/
- Time Commitment
  - [x] 4 Weeks
- Bounty Size
  - [x] 6000 USD

## Part 1: Problem Description

Mask offers a browser extension which lets users create encrypted tweets where the content of the tweet can only be “viewed” by other Twitter user if they use the Maskbook browser extension. These tweets can include text, but also images.

Unlock is a protocol for memberships which lets creators deploy a lock and sell access to this lock using the Ethereum blockchain. Locks can be added to websites, or applications in order to monetize them.

We are proposing an integration between Unlock and Mask where creators will be able to post encrypted Tweets and put them “behind” a lock, so that only members of the lock are able to view their content. If viewers are not members, then, Mask would offer the ability to Unlock these memberships. In that scenario, Mask is a referrer and should earn UDT.

For a more detailed description, you can refer to [this](https://docs.google.com/document/d/18bG3NAbfFZ7ofUEtq1Rdjs29XM1HveQ9htZ7rvC8JGo/edit?usp=sharing).

## Part 2: Resources

https://unlock-protocol.com/
https://github.com/unlock-protocol/unlock

## Part 3: Contact and Support

[Mask Discord Server](https://discord.gg/4Xccj7a75Q)
@yisiliu @Jack-Works -> Mask
@julien51 -> Unlock

### PR DESCRIPTION
**Pull request by guanbinrui**
Replacement of #3845.
closes #2363

### CODE REVIEW

File: packages/maskbook/src/plugins/UnlockProtocol/Services.ts
Comment (iamzubin) [inline packages/maskbook/src/plugins/UnlockProtocol/Services.ts:80]: Umm..... we need to remove this :3 
@guanbinrui

**Comment by yisiliu** [comment]
@guanbinrui do not merge yet since I've got a bug that's being investigated by @iamzubin

**Comment by iamzubin** [comment]
Umm, I made a PR to this branch, #4219 please take a look before merging this to the DEV

**Review by Jack-Works — APPROVED** [review]

**Pull request merged**
